python/dolfinx/wrappers/mesh.cpp

```diff
@@ -167,6 +167,26 @@ void declare_mesh(nb::module_& m, std::string type)
   std::string pyclass_geometry_name = std::string("Geometry_") + type;
   nb::class_<dolfinx::mesh::Geometry<T>>(m, pyclass_geometry_name.c_str(),
                                          "Geometry object")
+      .def(
+          "__init__",
+          [](dolfinx::mesh::Geometry<T>* self,
+             std::shared_ptr<const dolfinx::common::IndexMap> index_map,
+             nb::ndarray<std::int32_t, nb::ndim<2>, nb::c_contig> dofmap,
+             const dolfinx::fem::CoordinateElement<T>& element,
+             nb::ndarray<T, nb::ndim<2>, nb::c_contig> x, int dim,
```

Comment:
`x` should have shape `(num_points, dim)` and `dim` removed, or `x` should be constrained to have shape `(num_points, 3)`. At the moment it's neither. The former is simpler.